FetchContent fails to apply merge on submodule

During the build, while updating/refreshing some already built  submodules, we hit an auto-merging conflit  errors -
this usually happens, when the upstream source has been updated and FetContent need to pull and merge the changes locally.

This has the consequence to fail the entire build operation -

One way out is to delete the actual submodule so it can re-fetch and re-build

```sh

  Auto-merging test/lib/meta/plugin/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/meta/plugin/main.cpp

  Auto-merging test/lib/meta/plugin_std/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/meta/plugin_std/main.cpp

  Auto-merging test/lib/meta/plugin_std/userdata.h

  CONFLICT (add/add): Merge conflict in test/lib/meta/plugin_std/userdata.h

  Auto-merging test/lib/meta/shared/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/meta/shared/main.cpp

  Auto-merging test/lib/registry/plugin/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/registry/plugin/main.cpp

  Auto-merging test/lib/registry/shared/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/registry/shared/main.cpp

  Auto-merging test/lib/view/plugin/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/view/plugin/main.cpp

  Auto-merging test/lib/view/shared/main.cpp

  CONFLICT (add/add): Merge conflict in test/lib/view/shared/main.cpp

  Auto-merging testbed/CMakeLists.txt

  CONFLICT (add/add): Merge conflict in testbed/CMakeLists.txt

  error: could not apply b4e58bd...  *: updated TODO

  hint: Resolve all conflicts manually, mark them as resolved with

  hint: "git add/rm <conflicted_files>", then run "git rebase --continue".

  hint: You can instead skip this commit: run "git rebase --skip".

  hint: To abort and get back to the state before "git rebase", run "git
  rebase --abort".

  hint: Disable this message with "git config set advice.mergeConflict false"

  Could not apply b4e58bd...  # *: updated TODO





  You will have to resolve the conflicts manually



note: Run script build phase 'Generate CMakeFiles/ALL_BUILD' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'ALL_BUILD' from project 'entt-populate')
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution Generate\ entt-populate-prefix/src/entt-populate-stamp/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/entt-populate-update /Users/source/personals/RendererEngine/Result.Darwin.arm64.Debug/_deps/entt-subbuild/build/entt-populate.build/Debug/entt-populate.build/Script-20BD4BB91E55E18DEF6C2D1E.sh (in target 'entt-populate' from project 'entt-populate')
(1 failure)
CMake Error at /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1928 (message):
  Build step for entt failed: 65
Call Stack (most recent call first):
  /opt/homebrew/share/cmake/Modules/FetchContent.cmake:1619 (__FetchContent_populateSubbuild)
  /opt/homebrew/share/cmake/Modules/FetchContent.cmake:2155:EVAL:2 (__FetchContent_doPopulation)
  /opt/homebrew/share/cmake/Modules/FetchContent.cmake:2155 (cmake_language)
  /opt/homebrew/share/cmake/Modules/FetchContent.cmake:2394 (__FetchContent_Populate)
  dependencies.cmake:167 (FetchContent_MakeAvailable)
  CMakeLists.txt:44 (include)
```

--- a/dependencies.cmake
+++ b/dependencies.cmake
@@ -47,7 +47,7 @@ FetchContent_Declare(
   EnTT 
   GIT_REPOSITORY https://github.com/skypjack/entt.git
   GIT_SHALLOW TRUE 
-  GIT_TAG main
+  GIT_TAG v3.16.0
   )
 
 FetchContent_Declare(

